src/dashboard/apps/consent/views.py

```diff
@@ -33,11 +58,81 @@ def get_context_data(self, **kwargs):  # noqa: D102
         return context
 
 
+def _get_validate_control_authority() -> dict:
+    """Retrieves and validates the control authority details from settings.
+
+    Extracts the control authority information from settings.
+    Validates the extracted details according to the specified schema.
+    Returns them in a structured dictionary format.
+
+    Raises:
+    ------
+    ValidationError
+        If the control authority details fail the schema validation.
+    """
+    control_authority = {}
+    fields = ["name", "represented_by", "email"]
+    address_fields = {
+        "line_1": "address_1",
+        "line_2": "address_2",
+        "zip_code": "zip_code",
+        "city": "city",
+    }
+
+    for field in fields:
+        value = settings.CONSENT_CONTROL_AUTHORITY.get(field)
+        if value:
+            control_authority[field] = value
+
+    for key, setting_key in address_fields.items():
+        value = settings.CONSENT_CONTROL_AUTHORITY.get(setting_key)
+        if value:
+            control_authority.setdefault("address", {})[key] = value
```

Comment:
Why not changing the `CONSENT_CONTROL_AUTHORITY` schema instead? And only validate it?